src/test/java/it/eng/idsa/dataapp/handler/ArtifactMessageHandlerTest.java

```diff
@@ -31,15 +51,111 @@ public void init() {
 		assertNotNull(responseMap.get("payload"));
 		assertTrue(StringUtils.containsIgnoreCase(responseMap.get("header").toString(), message.getId().toString()));
 	}
+
+	@Test
+	void handleMessageTestRequestedArtifactNull() {
+
+		ArtifactRequestMessage arm = (ArtifactRequestMessage) message;
+		arm.setRequestedArtifact(null);
+
+		BadParametersException exception = assertThrows(BadParametersException.class, () -> {
+			responseMap = artifactMessageHandler.handleMessage(message, "asdsad");
+		});
+
+		assertEquals("Artifact requestedElement not provided", exception.getMessage());
+	}
+
+	@Test
+	void handleMessageRequestElementNotPresentInSelfDescription() {
+		
+		when(selfDescriptionService.getSelfDescription(message)).thenReturn(baseConnector);
+		when(selfDescriptionService.artifactRequestedElementExist((ArtifactRequestMessage) message,selfDescriptionService.getSelfDescription(message))).thenReturn(false);
+
+		NotFoundException exception = assertThrows(NotFoundException.class, () -> {
+			responseMap = artifactMessageHandler.handleMessage(message, "asdsad");
+		});
+		assertEquals("Artifact requestedElement not found in self description", exception.getMessage());
+	}
+
+	@Test
+	public void isBigPayloadTest() throws NoSuchMethodException, SecurityException, IllegalAccessException,
+			IllegalArgumentException, InvocationTargetException {
+
+		String pathWithBig = "http://w3id.org/engrd/connector/artifact/big";
+		String pathWithoutBig = "http://w3id.org/engrd/connector/artifact/1";
+		Method isBigPayload = ArtifactMessageHandler.class.getDeclaredMethod("isBigPayload", String.class);
+		isBigPayload.setAccessible(true);
+
+		assertTrue((boolean) isBigPayload.invoke(artifactMessageHandler, pathWithBig));
+		assertFalse((boolean) isBigPayload.invoke(artifactMessageHandler, pathWithoutBig));
+	}
+
+	@Test
+	public void createResponsePayloadTest() throws NoSuchMethodException, SecurityException, IllegalAccessException,
+			IllegalArgumentException, InvocationTargetException {
+
+		Method createResponsePayload = ArtifactMessageHandler.class.getDeclaredMethod("createResponsePayload");
+		createResponsePayload.setAccessible(true);
+
+		DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+		Date date = new Date();
+		String formattedDate = dateFormat.format(date);
+
+		Map<String, String> expectedPayload = new HashMap<>();
+		expectedPayload.put("firstName", "John");
+		expectedPayload.put("lastName", "Doe");
+		expectedPayload.put("dateOfBirth", formattedDate);
+		expectedPayload.put("address", "591  Franklin Street, Pennsylvania");
+		expectedPayload.put("checksum", "ABC123 " + formattedDate);
+
+		String expectedJson = new GsonBuilder().create().toJson(expectedPayload);
+		String actualJson = (String) createResponsePayload.invoke(artifactMessageHandler);
+
+		assertEquals(expectedJson, actualJson);
+	}
+
+	@Test
+	public void encodePayloadTest() throws NoSuchMethodException, SecurityException, IllegalAccessException,
```

Comment:
Why testing private method like this?
Can you create test for normal flow (public method) and then use parameter to call encode, and check result if encoded?
Same applies for other private method(s)

Reply:
Thank you for the suggestion.

It will be fixed in issue #53 since the whole issue is related to the code coverage.